extensions/common/json-ld/src/main/java/org/eclipse/edc/jsonld/transformer/TransformerUtil.java

```diff
@@ -0,0 +1,124 @@
+import static java.util.stream.Collectors.toList;
+import static org.eclipse.edc.jsonld.transformer.JsonLdKeywords.VALUE;
+
+public class TransformerUtil {
+    
+    private TransformerUtil() { }
+    
+    public static Object transformGenericProperty(JsonValue value, TransformerContext context) {
+        if (value instanceof JsonArray) {
+            var jsonArray = (JsonArray) value;
+            if (jsonArray.isEmpty()) {
+                return List.of();
+            } else if (jsonArray.size() == 1) {
+                // unwrap array
+                return context.transform(jsonArray.get(0), Object.class);
+            } else {
+                return jsonArray.stream().map(prop -> context.transform(prop, Object.class)).collect(toList());
+            }
+        } else {
+            return context.transform(value, Object.class);
+        }
+    }
+    
+    /**
+     * Transforms a JsonValue to a string and applies the result function. If the value parameter
+     * is not of type JsonString, JsonObject or JsonArray, a problem is reported to the context.
+     *
+     * @param value the value to transform
+     * @param resultFunction the function to apply to the transformation result
+     * @param context the transformer context
+     */
+    public static void transformString(JsonValue value, Consumer<String> resultFunction, TransformerContext context) {
+        if (value instanceof JsonString) {
+            var result = ((JsonString) value).getString();
+            resultFunction.accept(result);
+        } else if (value instanceof JsonObject) {
+            var jsonString = ((JsonObject) value).getJsonString(VALUE);
+            transformString(jsonString, resultFunction, context);
+        } else if (value instanceof JsonArray) {
+            transformString(((JsonArray) value).get(0), resultFunction, context);
+        } else {
+            context.reportProblem(format("Invalid property. Expected JsonString but got %s",
+                    value.getClass().getSimpleName()));
+        }
+    }
+    
+    /**
+     * Transforms a JsonValue to the desired output type and applies the result function. If the
+     * value parameter is not of type JsonObject, a problem is reported to the context.
+     *
+     * @param value the value to transform
+     * @param type the desired result type
+     * @param resultFunction the function to apply to the transformation result
+     * @param context the transformer context
+     * @param <T> the desired result type
+     */
+    public static <T> void transformObject(JsonValue value, Class<T> type, Consumer<T> resultFunction, TransformerContext context) {
+        if (value instanceof JsonObject) {
+            var result = context.transform((JsonObject) value, type);
+            resultFunction.accept(result);
+        } else if (value instanceof JsonArray) {
+            var array = (JsonArray) value;
+            transformObject(array.get(0), type, resultFunction, context);
```

Comment:
the array could be empty and throw an index out of bound exception here. Could be this method replaced with the `transformArrayOrObject` other than "only the first" logic, they are the same